src/pin.rs

```diff
@@ -25,11 +25,11 @@ where
         }
     }
 
-    pub(crate) fn pin_mask(&self) -> u32 {
+    pub fn pin_mask(&self) -> u32 {
         self.pin_mask
     }
 
-    pub(crate) fn port_driver(&self) -> &MUTEX {
+    pub fn port_driver(&self) -> &MUTEX {
         self.port_driver
     }
```

Comment:
Hm, I am not a fan of "leaking" the entire mutex out of the crate here.  This reveals implementation details of the internal mutex which means any internal mutex changes in the future will lead to breaking API changes.

Maybe it'd be more sensible to provide a method looking like this:

```rust
pub fn access_port_driver<F, R>(&self, f: F) -> R
where
    F: FnOnce(&mut PD) -> R,
{
    self.port_driver.lock(|pd| f(pd))
}
```